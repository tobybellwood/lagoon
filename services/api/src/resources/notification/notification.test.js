const { Sql } = require('./sql');

describe('Sql', () => {
  describe('selectAssignedProjectNotificationByName', () => {
    it('should create proper statement', () => {
      const input = { name: 'slack_foo', type: 'slack' };
      const ret = Sql.selectProjectNotificationByNotificationName(input);
      expect(ret).toMatchSnapshot();
    });
  });

  describe('selectNotificationsByTypeByProjectId', () => {
    it('should create a proper query', () => {
      const input = {
        type: 'slack',
        pid: 3,
      };

      const ret = Sql.selectNotificationsByTypeByProjectId(input);
      expect(ret).toMatchSnapshot();
    });
  });

  describe('updateNotificationSlack', () => {
    it('should create a proper query', () => {
      const input = {
        name: 'n1',
        patch: {
          name: 'test',
          webhook: 'new-webhook',
        },
      };

      const ret = Sql.updateNotificationSlack(input);
      expect(ret).toMatchSnapshot();
    });
  });

  describe('selectCustomer', () => {
    it('should create a proper query', () => {
      const ret = Sql.selectNotificationSlackByName('n1');
      expect(ret).toMatchSnapshot();
    });
  });

  describe('deleteProjectNotification', () => {
    it('should create a proper query', () => {
      const input = {
        project: 'some_project',
        notificationType: 'slack',
        notificationName: 'some_slack',
      };

      const ret = Sql.deleteProjectNotification(input);
      expect(ret).toMatchSnapshot();
    });
  });
});
