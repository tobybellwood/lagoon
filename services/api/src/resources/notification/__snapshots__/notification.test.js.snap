// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql deleteProjectNotification should create a proper query 1`] = `
"DELETE pn
      FROM project_notification as pn
      LEFT JOIN \`notification_slack\` AS nt ON pn.nid = nt.id AND pn.type = 'slack'
      LEFT JOIN project as p on pn.pid = p.id
      WHERE p.name = 'some_project'
      AND nt.name = 'some_slack'"
`;

exports[`Sql selectAssignedProjectNotificationByName should create proper statement 1`] = `"select \`nt\`.*, \`pn\`.*, 'slack' as type from \`project_notification\` as \`pn\` JOIN notification_slack AS nt ON pn.nid = nt.id AND pn.type = 'slack' where \`nt\`.\`name\` = 'slack_foo'"`;

exports[`Sql selectCustomer should create a proper query 1`] = `"select * from \`notification_slack\` where \`name\` = 'n1'"`;

exports[`Sql selectNotificationsByTypeByProjectId should create a proper query 1`] = `"select \`nt\`.*, \`pn\`.\`type\` from \`project_notification\` as \`pn\` JOIN notification_slack AS nt ON pn.nid = nt.id AND pn.type = 'slack' where \`pn\`.\`pid\` = 3"`;

exports[`Sql updateNotificationSlack should create a proper query 1`] = `"update \`notification_email\` set \`name\` = 'test', \`webhook\` = 'new-webhook' where \`name\` = 'n1'"`;
